app-libs/exchange-oracle/src/coingecko.rs

```diff
@@ -96,33 +122,87 @@ mod tests {
 	use super::*;
 	use core::assert_matches::assert_matches;
 
+	fn get_coingecko_coin_id(crypto_currency: &str) -> Result<String, Error> {
+		let url = CoinGeckoClient::base_url().unwrap();
+		let mut coingecko_client = CoinGeckoClient::new(url);
+		let trading_pair = TradingPair {
+			crypto_currency: crypto_currency.to_string(),
+			fiat_currency: "USD".to_string(),
+		};
+		coingecko_client.crypto_currency_id(trading_pair)
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_dot() {
+		let coin_id = get_coingecko_coin_id("DOT").unwrap();
+		assert_eq!(&coin_id, "polkadot");
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_teer() {
+		let coin_id = get_coingecko_coin_id("TEER").unwrap();
+		assert_eq!(&coin_id, "integritee");
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_ksm() {
+		let coin_id = get_coingecko_coin_id("KSM").unwrap();
+		assert_eq!(&coin_id, "kusama");
+	}
+
+	#[test]
+	fn crypto_currency_id_works_for_btc() {
+		let coin_id = get_coingecko_coin_id("BTC").unwrap();
+		assert_eq!(&coin_id, "bitcoin");
+	}
+
+	#[test]
+	fn crypto_currency_id_error_for_undefined_crypto_currency() {
+		let coin_id = get_coingecko_coin_id("Undefined");
+		assert_matches!(coin_id, Err(Error::InvalidCryptoCurrencyId));
+	}
+
 	#[test]
-	fn get_exchange_rate_for_undefined_coins_fails() {
+	fn get_exchange_rate_for_undefined_coingecko_crypto_currency_fails() {
 		let url = CoinGeckoClient::base_url().unwrap();
 		let mut coingecko_client = CoinGeckoClient::new(url);
-		let result = coingecko_client.get_exchange_rate("invalid_coin", "usd");
-		assert_matches!(result, Err(Error::NoValidData));
+		let trading_pair = TradingPair {
+			crypto_currency: "invalid_coin".to_string(),
+			fiat_currency: "USD".to_string(),
+		};
+		let result = coingecko_client.get_exchange_rate(trading_pair);
+		assert_matches!(result, Err(Error::InvalidCryptoCurrencyId));
 	}
 
 	#[test]
-	fn get_exchange_rate_for_undefined_currency_fails() {
+	fn get_exchange_rate_for_undefined_fiat_currency_fails() {
 		let url = CoinGeckoClient::base_url().unwrap();
 		let mut coingecko_client = CoinGeckoClient::new(url);
-		let result = coingecko_client.get_exchange_rate("polkadot", "ch");
+		let trading_pair =
+			TradingPair { crypto_currency: "DOT".to_string(), fiat_currency: "CH".to_string() };
+		let result = coingecko_client.get_exchange_rate(trading_pair);
 		assert_matches!(result, Err(Error::RestClient(_)));
 	}
 
 	#[test]
 	fn get_exchange_rate_from_coingecko_works() {
 		let url = CoinGeckoClient::base_url().unwrap();
 		let mut coingecko_client = CoinGeckoClient::new(url);
-		let dot_usd = coingecko_client.get_exchange_rate("polkadot", "usd").unwrap();
+		let dot_to_usd =
```

Comment:
I think that test is a little huge. Might make sense to split into multiple tests.

Reply:
It is one test, I can't split it. I can improve its structure and condense the code